processor/src/operations/decorators/mod.rs

```diff
@@ -7,9 +12,92 @@ impl Process {
     // DEBUGGING
     // --------------------------------------------------------------------------------------------
 
-    /// TODO: implement
-    pub fn op_debug(&self) -> Result<(), ExecutionError> {
-        unimplemented!();
+    /// Prints out debugging information based on options passed.
+    pub fn op_debug(&mut self, options: DebugOptions) -> Result<(), ExecutionError> {
+        info!(
+            "---------------------cycle: {}---------------------",
+            self.system.clk()
+        );
+        match options {
+            DebugOptions::All => {
+                self.print_stack(None);
+                self.print_mem(None, None);
+                self.print_local();
+            }
+            DebugOptions::Stack(n) => self.print_stack(n),
+            DebugOptions::Memory(n, m) => self.print_mem(n, m),
+            DebugOptions::Local(_) => self.print_local(),
+        }
+
+        Ok(())
+    }
+
+    // DEBUG HELPER
+    // ---------------------------------------------------------------------------------
+
+    /// Prints stack information for debugging.
+    /// If n is passed, this prints the top n items in the stack.
+    fn print_stack(&self, n: Option<usize>) {
+        let states = self.stack.get_values(n);
+        let values = states
+            .iter()
+            .map(|v| v.as_int().to_string())
+            .collect::<Vec<String>>()
+            .join(", ");
+        let depth = self.stack.depth();
+        info!("stack ({} of {}) ---------", n.unwrap_or(depth), depth);
+        info!("{}", values);
+    }
+
+    /// Create the display string for printing a word.
+    /// Prints <empty> if no value exists.
+    fn fmt_word(word: Option<Word>) -> String {
+        let shorten = |v: Felt| {
+            let bytes = v.as_int().to_le_bytes();
+            format!("{:#08x}..{:08x}", bytes[3], bytes[0])
```

Comment:
Wouldn't this print out only the 1st and the 4th bytes of an 8 byte value? And now that I think about it, to print out the whole value, we need 16 characters, which is exactly what we have now - 8 up front and 8 at the end. Maybe we should just print out the whole value in hex notation?